platform-hub-api/app/models/docker_repo.rb

```diff
@@ -34,7 +41,32 @@ class DockerRepo < ApplicationRecord
 
   validates :service_id, presence: true
 
+  validate_hashes(
+    access: {
+      schema: {
+        'robots' => [[
+          {
+            'username' => /\A[a-z]+[a-z0-9\-_]*\z/i,
```

Comment:
Just raising as a note, this and the repository `NAME_REGEX` is fine for ECR but may be different with other integrations depending on what they support.

Reply:
Yes, good point. Hopefully these are fairly conservative anyway, and thus forward compatible. But if not, we'll need to scope these checks by type.